crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -27,22 +29,58 @@ impl std::ops::Deref for Direction3d {
 }
 
 /// A sphere primitive
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq)]
 pub struct Sphere {
     /// The radius of the sphere
     pub radius: f32,
 }
 impl Primitive3d for Sphere {}
 
+impl Sphere {
+    /// Get the diameter of the sphere
+    pub fn diameter(&self) -> f32 {
+        2.0 * self.radius
+    }
+
+    /// Get the surface area of the sphere
+    pub fn area(&self) -> f32 {
+        4.0 * PI * self.radius.powi(2)
+    }
+
+    /// Get the volume of the sphere
+    pub fn volume(&self) -> f32 {
+        4.0 * FRAC_PI_3 * self.radius.powi(3)
+    }
+}
+
 /// An unbounded plane in 3D space. It forms a separating surface through the origin,
 /// stretching infinitely far
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq)]
 pub struct Plane3d {
     /// The normal of the plane. The plane will be placed perpendicular to this direction
     pub normal: Direction3d,
 }
 impl Primitive3d for Plane3d {}
 
+impl Plane3d {
+    /// Create a new `Plane3d` and get its translation from the origin based on three points.
+    /// The direction of the plane normal is determined by the winding order
+    /// of the triangle formed by the points.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `a == b`, `b == c` or `a == c`.
+    pub fn from_points(a: Vec3, b: Vec3, c: Vec3) -> (Self, Vec3) {
+        debug_assert!(
+            a != b && b != c && a != c,
+            "each corner of a triangle must be unique"
```

Comment:
I changed it to be "plane must be defined by three finite points that don't lie on the same line". "non-collinear" would probably be fine too, but I hadn't heard it before and had to Google it, so I think it's better to favor more explicative language for people with less knowledge on math terminology (or who have learned it in a different language, like me).

I also made the assert use `.is_normalized()` instead since that's what really should check if the normal is valid